app/src/main/java/com/twilio/conversations/app/manager/MessageListManager.kt

```diff
@@ -74,23 +74,25 @@ class MessageListManagerImpl(
             uuid
         )
         conversationsRepository.insertMessage(message)
-        val sentMessage = conversation.sendMessage(options).toMessageDataItem(identity, uuid)
+
+        val unsentMessage = conversation.prepareMessage().setBody(text).setAttributes(attributes).build()
+        val sentMessage = unsentMessage.sendMessage().toMessageDataItem(identity, uuid)
         conversationsRepository.updateMessageByUuid(sentMessage)
     }
 
     override suspend fun retrySendTextMessage(messageUuid: String) {
         val message = withContext(dispatchers.io()) { conversationsRepository.getMessageByUuid(messageUuid) } ?: return
         if (message.sendStatus == SendStatus.SENDING.value) return
+
         conversationsRepository.updateMessageByUuid(message.copy(sendStatus = SendStatus.SENDING.value))
+
         val identity = conversationsClient.getConversationsClient().myIdentity
         val attributes = Attributes(message.uuid)
-        val options = Message.options().withBody(message.body).withAttributes(attributes)
-        val sentMessage = conversationsClient
-            .getConversationsClient()
-            .getConversation(conversationSid)
-            .sendMessage(options)
-            .toMessageDataItem(identity, message.uuid)
 
+        val conversation = conversationsClient.getConversationsClient().getConversation(conversationSid)
+
+        val unsentMessage = conversation.prepareMessage().setBody(message.body).setAttributes(attributes).build()
+        val sentMessage = unsentMessage.sendMessage().toMessageDataItem(identity, message.uuid)
```

Comment:
why not all at once?

```
val sentMessage = conversation.prepareMessage()
    .setBody(message.body)
    .setAttributes(attributes)
    .buildAndSend()
    .toMessageDataItem(identity, message.uuid)
```

Reply:
It wouldn't compile.
`buildAndSend()` returns `CancellationToken` and takes listener `CallbackListener<Message>`.

Reply:
Added `buildAndSend()` extension. 
With 3.0.1, will switch to extension from SDK.

Reply:
Now using buildAndSend() extension